src/plugins/foxPage/foxPage.tsx

```diff
@@ -41,29 +47,31 @@ export const FoxPage = (props: FoxPageProps) => {
   const history = useHistory()
   const assetFox = useAppSelector(state => selectAssetById(state, FoxAssetId))
   const assetFoxy = useAppSelector(state => selectAssetById(state, FoxyAssetId))
-  const isFoxSelected = props.activeAssetId === FoxAssetId
-  const isFoxySelected = props.activeAssetId === FoxyAssetId
-  const selectedAsset = isFoxSelected ? assetFox : assetFoxy
+
+  const assets = useMemo(() => [assetFox, assetFoxy], [assetFox, assetFoxy])
+
+  const selectedAssetIndex = useMemo(
+    () => assets.findIndex(asset => asset.assetId === props.activeAssetId),
+    [props.activeAssetId, assets],
+  )
+
+  const selectedAsset = useMemo(
+    () => assets.find(asset => asset.assetId === props.activeAssetId),
+    [props.activeAssetId, assets],
+  )
```

Comment:
Nitpick, that improves the time complexity from O(n) to O(1), not a big deal because of memoization and the fact that it's just an array of two, but still nicer to not see repeated iterations that could be avoided
```suggestion
  const selectedAsset = useMemo(
    () => assets[selectedAssetIndex],
    [selectedAssetIndex, assets],
  )
```

Reply:
Actually memoizing won't even be needed anymore here, since `assets` is coming from a selector and is referentially stable because of the memoization, and accessing an element by index here will also give us a stable reference/is a cheap O(1) operation